### DISCUSSION TITLE
Change in behaviour of DiracSimplify

### DISCUSSION THREAD

**Discussion by NefedovM** [FeynCalc/feyncalc#235] [Q&A]
Hi All,
I just have re-installed the development version 10.0.0 of the FeynCalc using the usual command from the web-site and immediately ran into a change in behaviour of DiracSimplify/DiracTrace which might confuse old FeynCalc users such as me.

Before I also had a development version 10.0.0 installed, which perhaps dates back from January, but I have decided to re-install it today. In the old version the sequence of commands:
 
```
FCI[DiracTrace[GA[\[Mu]] . GA[\[Nu]]]]
% /. DiracTrace -> Tr
```

would lead to the computation of the trace, i.e. 4*g_{mu,nu} . Now it does nothing. Instead, doing DiracSimplify of the expression computes the trace now. 

Before it used to be different. DiracSimplify typically did the simplifications on Dirac algebra but did not compute the traces and to actually compute the trace one had to do the replacement  DiracTrace -> Tr . This behaviour is assumed in lot of my codes and now I have to be careful with this. 

So my question is: was it intentional or is it a bug or I am mixing up something?

All the best,
Maxim

**Comment by vsht** [comment]
Hi Maxim,

it's true, that `Tr` does not evaluate traces anymore. It was intentional, 

https://github.com/FeynCalc/feyncalc/commit/13be92a1fd24774d7c41477a07d7d4608c4c2dcb

because `Tr` is actually a built-in Mathematica function that has nothing to do with FeynCalc. The old behavior
was achieved by overloading this function and changing its definition accordingly. However, modifying the behavior
of built-in function is usually considered to be a very poor Mathematica programming practice that can easily backfire
in many unexpected ways. This is why I decided that the next FeynCalc release is a good moment to get rid of such
dirty tricks.

Having said that, the old behavior can be easily recovered by just evaluating
```
Unprotect[Tr];
Tr[trarg_] := 
  TR[trarg] /; ! FreeQ[{trarg}, FeynCalc`Package`TrFeynCalcObjects];
Tr /: Options[Tr] := Options[TR];
Protect[Tr];
```
Cheers,
Vladyslav
